make-pangenome-sketches.py

```diff
@@ -21,52 +23,88 @@ def main():
     )
     p.add_argument('--scaled', default=1000, type=int)
     p.add_argument('-k', '--ksize', default=31, type=int)
-    p.add_argument('-o', '--output', required=True)
+    p.add_argument('-o', '--output', required=True,
+                    help='Define a filename for the pangenome signatures (.zip preferred). \n'
+                         'A CSV file with the same name will also be generated to contain the lineage rank, genome name, and hash count.')
     p.add_argument('-r', '--rank', default='species')
+    p.add_argument('-a', '--abund', action='store_true', help='Enable abundance tracking of hashes across rank selection.')
     args = p.parse_args()
 
     print(f"loading taxonomies from {args.taxonomy_file}")
     taxdb = sourmash.tax.tax_utils.MultiLineageDB.load(args.taxonomy_file)
     print(f"found {len(taxdb)} identifiers in taxdb.")
 
-    revtax_d = {}
     ident_d = {}
+    revtax_d = {}
+    accum = defaultdict(dict)
+    if args.abund:
+        counts = {}
+    csv_file = check_csv(args.output)
 
+    # Load the database
     for filename in args.sketches:
         print(f"loading file {filename} as index => manifest")
         db = sourmash_args.load_file_as_index(filename)
         db = db.select(ksize=args.ksize)
         mf = db.manifest
         assert mf, "no matching sketches for given ksize!?"
 
-        # load and merge
+        chunk = []
+        # Work on a single signature at a time across the database
         for n, ss in enumerate(db.signatures()):
             if n and n % 1000 == 0:
                 print(f'...{n} - loading')
-            name = ss.name
-            md5sum = ss.md5sum()
 
+            name = ss.name
             ident = tax_utils.get_ident(name)
-
+            
             # grab relevant lineage name
             lineage_tup = taxdb[ident]
             lineage_tup = tax_utils.RankLineageInfo(lineage=lineage_tup)
             lineage_pair = lineage_tup.lineage_at_rank(args.rank)
             lineage_name = lineage_pair[-1].name
 
             # pick an ident to represent this set of pangenome sketches
-            ident_d[lineage_name] = ident # ok if overwrite...
+            ident_d[lineage_name] = ident 
+
+            # Accumulate the count within lineage names if `--abund` in cli
+            if args.abund:
+                c = Counter(ss.minhash.hashes)
+                if lineage_name in counts:
+                    counts[lineage_name].update(c)
+                else:
+                    counts[lineage_name] = c
 
             # track merged sketches
             mh = revtax_d.get(lineage_name)
             if mh is None:
                 mh = ss.minhash.to_mutable()
+            
```

Comment:
lotsa extra whitespace being added...

Reply:
I am struggling with the line between readability and line count. What is your guidance on this topic, o' great PI?

Reply:
Oh, okay. I'll see if I can coax vim into showing me all the whitespaces in a file. Thanks!

Reply:
`:set list`!